liberty-maven-plugin/src/main/java/net/wasdev/wlp/maven/plugins/server/DevMojo.java

```diff
@@ -0,0 +1,889 @@
+                    }
+
+                    log.info("Running integration tests...");
+                    try {
+                        runIntegrationTests();
+                        log.info("Integration tests finished.");
+                    } catch (MojoExecutionException e) {
+                        Throwable cause = e.getCause();
+                        if (cause != null && cause instanceof MojoFailureException) {
+                            log.debug(e);
+                            log.error("Integration tests failed: " + cause.getLocalizedMessage());
+                        } else {
+                            log.error("Failed to run integration tests", e);
+                        }
+                    }
+                }
+            }
+
+            // finally, start watching for hotkey presses if not already started
+            util.runHotkeyReaderThread(executor);
+        }
+
+        @Override
+        public void checkConfigFile(File configFile, File serverDir) {
+            try {
+                ServerFeature servUtil = new ServerFeature();
+                Set<String> features = servUtil.getServerFeatures(serverDir);
+                features.removeAll(existingFeatures);
+                if (!features.isEmpty()) {
+                    List<String> configFeatures = new ArrayList<String>(features);
+                    log.info("Configuration features have been added");
+                    runMojo("net.wasdev.wlp.maven.plugins:liberty-maven-plugin", "install-feature", serverName,
+                            configFeatures);
+                    this.existingFeatures.addAll(features);
+                }
+            } catch (Exception e) {
+                log.debug("Failed to read configuration file", e);
+            }
+        }
+
+        @Override
+        public boolean compile(File dir) {
+            try {
+                if (dir.equals(sourceDirectory)) {
+                    log.info("Running maven-compiler-plugin:compile");
+                    runMojo("org.apache.maven.plugins:maven-compiler-plugin", "compile", null, null);
+
+                    log.info("Running maven-compiler-plugin:resources");
+                    runMojo("org.apache.maven.plugins:maven-resources-plugin", "resources", null, null);
+                }
+                if (dir.equals(testSourceDirectory)) {
+                    log.info("Running maven-compiler-plugin:testCompile");
+                    runMojo("org.apache.maven.plugins:maven-compiler-plugin", "testCompile", null, null);
+                    log.info("Running maven-compiler-plugin:testResources");
+                    runMojo("org.apache.maven.plugins:maven-resources-plugin", "testResources", null, null);
+                }
+                return true;
+            } catch (MojoExecutionException e) {
+                log.error("Unable to compile", e);
+                return false;
+            }
+        }
+    }
+
+    DevMojoUtil util;
```

Comment:
This shouldn't need to be at the class level right?

Reply:
Moved it into the doExecute() method in https://github.com/WASdev/ci.maven/pull/476/commits/acdecd3805337ec4a3c1f457c231d6a001aaddd2